src/contract/ga/index.ts

```diff
@@ -0,0 +1,171 @@
+ *
+ * @module @aeternity/aepp-sdk/es/contract/ga
+ * @export GeneralizedAccount
+ * @example import { GeneralizedAccount } from '@aeternity/aepp-sdk'
+ */
+
+// @ts-expect-error TODO: remove me
+import { TX_TYPE } from '../../tx/builder/schema'
+// @ts-expect-error TODO: remove me
+import { buildTx, unpackTx } from '../../tx/builder'
+import { prepareGaParams } from './helpers'
+import { hash } from '../../utils/crypto'
+import { decode, EncodedData } from '../../utils/encoder'
+// @ts-expect-error TODO remove
+import { getNetworkId } from '../../node'
+import { IllegalArgumentError, MissingParamError } from '../../utils/errors'
+
+export interface AuthData {
+  gasLimit: number
+  callData: EncodedData<'cb'>
+  source?: string
+  args?: object
+}
+
+interface GAObject {
+  owner: string
+  transaction: string
+  rawTx: string
+  gaContractId: string
+}
+
+/**
+ * @alias module:@aeternity/aepp-sdk/es/contract/ga
+ * Check if account is GA
+ * @param address - Account address
+ * @returns if account is GA
+ */
+export async function isGA ({ onAccount }: {onAccount: any}, address: string): Promise<boolean> {
+  const { contractId } = await onAccount.getAccount(address)
+  return contractId != null
+}
+
+/**
+ * Convert current account to GA
+ * @alias module:@aeternity/aepp-sdk/es/contract/ga
+ * @param authFnName - Authorization function name
+ * @param source - Auth contract source code
+ * @param args - init arguments
+ * @param options - Options
+ * @returns General Account Object
+ */
+export async function createGeneralizedAccount (
+  { onAccount }: { onAccount: any},
+  authFnName: string,
+  source: string,
+  args: any[] = [],
+  options: object = {}):
+  Promise<Readonly<GAObject>> {
+  const opt = { ...onAccount.Ae.defaults, ...options }
+  const ownerId: string = await onAccount.address(opt)
+  if (await isGA({ onAccount }, ownerId)) throw new IllegalArgumentError(`Account ${ownerId} is already GA`)
+
+  const contract = await onAccount.getContractInstance({ onAccount, source })
+
+  await contract.compile()
+  const { tx, contractId }: {tx: string, contractId: string} = await onAccount.gaAttachTx({
+    ...opt,
+    gasLimit: opt.gasLimit ?? await contract._estimateGas('init', args, opt),
+    ownerId,
+    code: contract.bytecode,
+    callData: contract.calldata.encode(contract._name, 'init', args),
+    authFun: hash(authFnName)
+  })
+
+  const { hash: transaction, rawTx } = await onAccount.send(tx, opt)
+
+  return Object.freeze({
+    owner: ownerId,
+    transaction,
+    rawTx,
+    gaContractId: contractId
+  })
+}
+
+/**
+ * Create a metaTx transaction
+ * @alias module:@aeternity/aepp-sdk/es/contract/ga
+ * @param rawTransaction Inner transaction
+ * @param authData Object with gaMeta params
+ * @param authFnName - Authorization function name
+ * @param options - Options
+ * @returns Transaction string
+ */
+export async function createMetaTx (
+  { onAccount }: { onAccount: any },
```

Comment:
This should be moved into the last `options` argument

Reply:
The approach you’re following with chain (putting `onAccount` in the `options` param) doesn’t seem to be working in the case of `ContractMethods` and `GeneralAccountMethods`. Probably because we don’t always pass options to some of those functions or because of the `= {}` initializer. If we put the `onAccount` argument in the `options` param, without passing `options` when calling the function, `onAccount` will end up somewhere else. This is why I put `onAccount` at the top of the parameters. This way it’s always where we expect it to be.

Reply:
`onAccount` shouldn't be in the first argument, make all arguments before options required then you won't have propblem with detecting the options object